lib/js/components/Layout/Layout.vue

```diff
@@ -0,0 +1,261 @@
+<template>
+	<div class="layout">
+		<div v-if="sideMenuVisible" class="layout__navigation">
+			<div class="layout__mainMenu">
+				<slot name="mainMenu" />
+			</div>
+			<div class="layout__sideNav">
+				<slot name="sideNav" />
+			</div>
+		</div>
+		<div
+			class="layout__contentColumn"
+			:class="{
+				'-fullWidth': isContentFullWidth,
+				'-max900': !isContentFullWidth,
+				'-noPadding': contentWithoutPadding,
+			}"
+		>
+			<slot />
+		</div>
+		<template
+			v-if="
+				$slots.rightColumn &&
+				rightColumnVisible &&
+				!isContentFullWidth &&
+				rightColumnMode === LAYOUT_RIGHT_COLUMN_MODE.COLUMN_VISIBLE
+			"
+		>
+			<div class="layout__overlay" />
+			<div
+				class="layout__rightColumn"
+				:class="{
+					'-medium': rightColumnSize === LAYOUT_RIGHT_COLUMN_SIZE.MEDIUM,
+					'-large': rightColumnSize === LAYOUT_RIGHT_COLUMN_SIZE.LARGE,
+				}"
+			>
+				<slot name="rightColumn" />
+			</div>
+		</template>
+		<div
+			v-if="
+				$slots.sideBar &&
+				sideBarVisible &&
```

Comment:
I'm not sure if we agreed with UX that collapsed right collumn will be rendered using separate slot or different way. Do you have a confirmation for that?